gemd/entity/value/uniform_integer.py

```diff
@@ -17,10 +17,35 @@ class UniformInteger(IntegerValue):
 
     typ = "uniform_integer"
 
-    def __init__(self, lower_bound=None, upper_bound=None):
-        assert isinstance(lower_bound, int)
-        assert isinstance(upper_bound, int)
-        assert lower_bound <= upper_bound, \
-            "the lower bound must be <= the upper bound"
+    def __init__(self, lower_bound: int, upper_bound: int):
+        self._lower_bound = None
+        self._upper_bound = None
+
         self.lower_bound = lower_bound
         self.upper_bound = upper_bound
+        assert self.lower_bound <= self.upper_bound, \
+            "the lower bound must be <= the upper bound"
+
+    @property
+    def lower_bound(self) -> int:
+        """The lower bound of a uniform distribution."""
+        return int(self._lower_bound)
+
+    @lower_bound.setter
+    def lower_bound(self, lower_bound: int) -> None:
+        """The lower bound of a uniform distribution."""
+        # This check is necessary to handle JSON serialization behavior under 3.5
+        assert float(int(lower_bound)) == float(lower_bound), "lower bound must be an int"
+        self._lower_bound = int(lower_bound)
+
+    @property
+    def upper_bound(self) -> int:
+        """The upper bound of a uniform distribution."""
+        return int(self._upper_bound)
+
+    @upper_bound.setter
+    def upper_bound(self, upper_bound: int) -> None:
+        """The upper bound of a uniform distribution."""
+        # This check is necessary to handle JSON serialization behavior under 3.5
+        assert float(int(upper_bound)) == float(upper_bound), "upper bound must be an int"
```

Comment:
I think if `upper_bound` is a string, this will throw a ValueError due to `int("foo")` rather than an assertion error and not return this error message.

Reply:
That is correct.  But you think a ValueError would be more appropriate here than a TypeError?

Reply:
I care most about the message and second most about consistency.  No strong opinions about the type of error.